### ISSUE TITLE
[BUG] let vs vars if is-variable is evaluated in a filter

### ISSUE THREAD

**Issue by pmario** [Jermolene/TiddlyWiki5#7348]
### Describe the bug

To reproduce $let-widget does the wrong thing  -- $vars works as expected

let-widget always shows aa even if "default" is _not_ a variable

```
<$let default={{{ [[default]is[variable]then[aa]else[bb]] }}} >
<<default>>
</$let>

<$vars default={{{ [[default]is[variable]then[aa]else[bb]] }}} >
<<default>>
</$vars>
```

### Screenshots

![image](https://user-images.githubusercontent.com/374655/224071683-063026a0-007d-405d-9c87-14ce581aad67.png)

### TiddlyWiki Configuration

v5.2.5-prerelease

### Additional context

Related to: [BUG] editions/server-external-js is broken #7346 

The following code will always set coreURL to an empty string if coreURL does not exist

https://github.com/Jermolene/TiddlyWiki5/blob/b16d6953863c144bcae63374bf779bcb6a6bc78b/core/templates/external-js/save-offline-external-js.tid#L12

**Comment by pmario** [comment]
@Jermolene .. I did update the OP .. had a copy paste problem

**Comment by kookma** [comment]
Hi Mario,
For me it works with no error. TW 5.2.5

```
\define default() hi

<$let default={{{ [[default]is[variable]then[aa]else[bb]] }}} >
<<default>>
</$let>

<$vars default={{{ [[default]is[variable]then[aa]else[bb]] }}} >
<<default>>
</$vars>
```

produces

aa

aa

**Comment by pmario** [comment]
That's what I wrote `\define default() hi` is _not_ allowed

**Comment by Jermolene** [comment]
I think @pmario is pointing out that there is an inconsistency if the variable "default" is not defined:

```
<$let default={{{ [[default]is[variable]then[aa]else[bb]] }}} >
<<default>>
</$let>

<$vars default={{{ [[default]is[variable]then[aa]else[bb]] }}} >
<<default>>
</$vars>
```

The result in v5.2.6 is certainly anomalous:

```
aa

bb
```

What is going on is that the let widget allows assignments to reference the values of variables that are set by previous attributes on the same let widget. This is what makes it possible to swap two variables like this:

```
\define showvalues() (<$text text=<<a>>/>|<$text text=<<b>>/>)

<$let a="elephant" b="giraffe">

<<showvalues>>

<$let tmp=<<a>> a=<<b>> b=<<tmp>>>

<<showvalues>>

</$let>

<<showvalues>>

</$let>
```

The output of that example is:

```
(elephant|giraffe)

(giraffe|elephant)

(elephant|giraffe)
```

I think that where the bug comes in is that the current implementation reports a value for all the variables set by the widget, regardless of whether they have yet have an assignment made.

**Comment by saqimtiaz** [comment]
> I think that where the bug comes in is that the current implementation reports a value for all the variables set by the widget, regardless of whether they have yet have an assignment made.

[This change](https://github.com/Jermolene/TiddlyWiki5/commit/8c378e0d24e9f4b8986316bc17f077b894d1eb30) seems to be the cause of the change in behaviour, since the `is[variable]` operator calls `widget.getVariable()` which now returns a blank string for undefined variables.

**Comment by Jermolene** [comment]
Thank you @saqimtiaz I'd just got to the same place!

**Comment by Jermolene** [comment]
Fixed in 007b75942166e1ff138a46893250d4e62e85c85d
